Bugfix in LeptonSF modules and change naming for analysis recipes

squinto5 opened the pull request:
Spotted a bug in LeptonSF module, probably affect whole production.
Also changing naming for framework tests.

NTrevisani commented on mkShapesRDF/processor/framework/samples/samplesCrossSections_13p6TeV.py:
```diff
@@ -124,11 +124,11 @@
 
 # W+Jets
-xs_db["WToLNu-2Jets"] = ["xsec=67710.0", "kfact=1.000", "ref=G"]
+xs_db["WToLNu-2Jets"] = ["xsec=63396.0", "kfact=1.000", "ref=G"]
```
Could you update the reference to MATRIX? 
--> "ref=A"

NTrevisani commented on mkShapesRDF/shapeAnalysis/latinos/PlotFactory.py:
```diff
@@ -1815,7 +1815,7 @@ def makePlot(
                 CMS_lumi.lumi_8TeV = "18.3 fb^{-1}"
                 CMS_lumi.lumi_13TeV = "100 fb^{-1}"
                 CMS_lumi.writeExtraText = 1
```
is it the current recommendation for PAS-like plots?

NTrevisani commented on mkShapesRDF/shapeAnalysis/latinos/PlotFactory.py:
```diff
@@ -1815,7 +1815,7 @@ def makePlot(
                 CMS_lumi.lumi_13TeV = "100 fb^{-1}"
                 CMS_lumi.writeExtraText = 1
-                CMS_lumi.extraText = "Preliminary"
+                CMS_lumi.extraText = "Private work"
```
Is this the latest recommendation for PAS-like plots?

squinto5 commented:
@BlancoFS maybe you know better what should be done here https://github.com/latinos/mkShapesRDF/blob/squinto_dev/mkShapesRDF/processor/data/LeptonSel_cfg.py#L769

squinto5 commented:
> > Hi @NTrevisani, with the latest commit I've changed everywhere (I think) the correction set used from latest to the last date possible (i.e. latest -> 2025-12-03) to ensure a stable production and consistency. On top of that I've changed the necessary keys. Last, I have a doubt about this line: https://github.com/latinos/mkShapesRDF/pull/81/files#diff-23905ba2f59ff46deac702881f67af584c7dde47e0dafacf5fc6a6fd3722680dR769
> 
> Excellent! Do we need any other changes or is it all?

I think it is all, hope everything will run smoothly this time

BlancoFS commented:
> @BlancoFS maybe you know better what should be done here https://github.com/latinos/mkShapesRDF/blob/squinto_dev/mkShapesRDF/processor/data/LeptonSel_cfg.py#L769

Ciao @squinto5, I think it's good as it's now. So, I would just remove the comment. There is one last point that can be improved. There are still some missing scale factors for the latest working point that I introduced (loose tthMVA for electrons with pt<20 GeV) for 2022 and 2023. The ones for 2024 are already there. I have already computed them. If you want, I can share the new json files with you so you can upload them to this PR. It corresponds to these lines, for example:

https://github.com/latinos/mkShapesRDF/blob/bf6dca6eb3432ce714d3534f03e659d3d2e6985c/mkShapesRDF/processor/data/LeptonSel_cfg.py#L679

squinto5 commented:
@NTrevisani @BlancoFS I've updated the PR, please check the last commit if everything checks out

BlancoFS commented:
I agree with Nicolò, and also don't think it's the most correct way to do it. Would it make sense to leave it as something to be done at analysis level?

BlancoFS commented on mkShapesRDF/processor/modules/JetSelMask.py:
```diff
@@ -56,37 +92,37 @@ def runModule(self, df, values):
-            print("Mask applied")
+            df = df.Define("Jet_jetId", "Jet_ID(Jet_eta,Jet_chHEF,Jet_neHEF,Jet_chEmEF,Jet_neEmEF,Jet_muEF,Jet_chMultiplicity,Jet_neMultiplicity)")
+
+        jetIdCut = f"Take(Jet_jetId, CleanJet_jetIdx) >= {self.jetId}"
```
Aaaah ok I forgot that the TightLeptonVeto was only for veto-maps. For me, it's fine loke that. Good, then it's good for me and I don't have more comments

NTrevisani replied:
TightLeptonVeto removes jets overlapping with leptons? I guess this is what we want also on CleanJets. Am I right?
